runtime_scan/pkg/provider/docker/client.go

```diff
@@ -0,0 +1,389 @@
+	}
+
+	// Get container assets
+	containerAssets, err := c.getContainers(ctx)
+	if err != nil {
+		return nil, provider.FatalErrorf("failed to get containers. Provider=%s: %w", models.Docker, err)
+	}
+
+	// Combine assets
+	assets := append(imageAssets, containerAssets...)
+
+	return assets, nil
+}
+
+func (c *Client) RunAssetScan(ctx context.Context, config *provider.ScanJobConfig) error {
+
+	err := c.prepareScanVolume(ctx, config)
+	if err != nil {
+		return provider.FatalErrorf("failed to prepare scan volume. Provider=%s: %w", models.Docker, err)
+	}
+
+	err = c.createScanConfigFile(config)
+	if err != nil {
+		return provider.FatalErrorf("failed to create scanconfig.yaml file. Provider=%s: %w", models.Docker, err)
+	}
+
+	containerId, err := c.createScanContainer(ctx, config)
+	if err != nil {
+		return provider.FatalErrorf("failed to create scan container. Provider=%s: %w", models.Docker, err)
+	}
+
+	err = c.dockerClient.ContainerStart(ctx, containerId, types.ContainerStartOptions{})
+	if err != nil {
+		return provider.FatalErrorf("failed to start scan container. Provider=%s: %w", models.Docker, err)
+	}
+
+	return nil
+}
+
+func (c *Client) RemoveAssetScan(ctx context.Context, config *provider.ScanJobConfig) error {
+	scanName, err := getScanName(config)
+	if err != nil {
+		return provider.FatalErrorf("failed to get scan name. Provider=%s: %w", models.Docker, err)
+	}
+
+	scanConfigFileName, err := getScanConfigFileName(config)
+	if err != nil {
+		return provider.FatalErrorf("failed to get scan config file name. Provider=%s: %w", models.Docker, err)
+	}
+	err = os.Remove(path.Dir(scanConfigFileName))
+	if err != nil {
+		return provider.FatalErrorf("failed to remove scan config file. Provider=%s: %w", models.Docker, err)
+	}
+
+	containerId, err := c.getContainerIdFromContainerName(ctx, scanName)
+	if err != nil {
+		return provider.FatalErrorf("failed to get scan container id. Provider=%s: %w", models.Docker, err)
+	}
+	err = c.dockerClient.ContainerRemove(ctx, containerId, types.ContainerRemoveOptions{Force: true})
+	if err != nil {
+		return provider.FatalErrorf("failed to remove scan container. Provider=%s: %w", models.Docker, err)
+	}
+
+	err = c.dockerClient.VolumeRemove(ctx, scanName, true)
+	if err != nil {
+		return provider.FatalErrorf("failed to remove volume. Provider=%s: %w", models.Docker, err)
+	}
+
+	networkId, err := c.getNetworkIdFromNetworkName(ctx, scanName)
+	if err != nil {
+		return provider.FatalErrorf("failed to get scan network id. Provider=%s: %w", models.Docker, err)
+	}
+	err = c.dockerClient.NetworkRemove(ctx, networkId)
+	if err != nil {
+		return provider.FatalErrorf("failed to remove scan network. Provider=%s: %w", models.Docker, err)
+	}
+
+	return nil
+}
+
+func (c *Client) prepareScanVolume(ctx context.Context, config *provider.ScanJobConfig) error {
+	logger := log.GetLoggerFromContextOrDiscard(ctx)
+
+	scanName, err := getScanName(config)
```

Comment:
Why not using the AssetScanID (we have it in the config.Metadata)

Reply:
agree 👍